src/pages/cli/project/troubleshooting.mdx

```diff
@@ -0,0 +1,191 @@
+1. Amplify CLI input parameters artifacts: All user provided inputs are stored in JSON files in the `/amplify/backend` folder. For example, `cli-inputs.json` or `amplify-meta.json`. These files let you inspect the configurations, both developer-provided and Amplify-generated, to root cause potential problems. Refer to [Amplify Backend Files](https://docs.amplify.aws/cli/reference/files/#backend-configjson) to determine if a file is safe to edit manually.
+
+### Amplify GraphQL API
+Resolving deployment issues in Amplify projects with a GraphQL API:
+
+#### Understanding Amplify GraphQL artifacts:
+
+The Amplify CLI GraphQL workflow helps generate AWS AppSync infrastructure resources (API definitions, schema, resolvers) and client-side code. Running `amplify api gql-compile` and [amplify codegen](https://docs.amplify.aws/cli/graphql/client-code-generation/) prior to running `amplify push` may help developers to validate their GraphQL schemas and avoid getting `amplify push` failures. Please refer to the [Amplify GraphQL Troubleshooting guide](https://docs.amplify.aws/cli/graphql/troubleshooting/).
+  * GraphQL schema VTL generation: The Amplify CLI GraphQL workflow uses the [AWS AppSync service](https://docs.aws.amazon.com/appsync/latest/devguide/system-overview-and-architecture.html) to provision the GraphQL API. The “amplify api gql-compile” command transpiles the GraphQL schema provided by the developer and generates all the artifacts required to deploy the API in AWS. 
+    +	Directives like [@model](https://docs.amplify.aws/cli/graphql/data-modeling/), [@function](https://docs.amplify.aws/cli-legacy/graphql-transformer/function/), [@auth](https://docs.amplify.aws/cli-legacy/graphql-transformer/auth/#auth) and [@searchable](https://docs.amplify.aws/cli-legacy/graphql-transformer/searchable/#definition) in the GraphQL schema are used to generate CloudFormation and provision AWS resources.
+    +	The GraphQL resolvers for Query, Mutation or Subscription are converted into [VTL (Velocity Template Language)](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-mapping-template-reference-programming-guide.html) files. AWS AppSync uses VTL to translate GraphQL requests from clients into a request to your data source.
+
+  * GraphQL schema client-side code generation: The Amplify CLI GraphQL workflow allows developers to generate client side code for web and mobile clients using the `amplify codegen` command. 
+
+## Troubleshooting `amplify push` failures
```

Comment:
```suggestion
## Troubleshooting "amplify push" failures
```

this is a weird one. The rendering has problems with code blocks in headers... so converting to quotation marks.
![image](https://user-images.githubusercontent.com/4989523/152068880-77dcc442-47a3-4c93-8593-f04e5c6bede0.png)